common/lib/client/realtimechannel.js

```diff
@@ -49,6 +49,57 @@ var RealtimeChannel = (function() {
 		return args;
 	};
 
+	RealtimeChannel.prototype.setOptions = function(options, callback) {
+		if(!callback) {
+			if (this.rest.options.promises) {
+				return Utils.promisify(this, 'setOptions', arguments);
+			}
+
+			callback = (err) => {
+				if(err) {
+					Logger.logAction(Logger.LOG_MAJOR, 'RealtimeChannel.setOptions()', 'Set options failed: ' + err.toString());
+				}
+			};
+		}
+		var err = validateChannelOptions(options);
+		if(err) {
+			callback(err);
+			return;
+		}
+		Channel.prototype.setOptions.call(this, options);
+		if(this.shouldReattachToSetOptions(options)) {
+			this._attach(true, callback);
+		} else {
+			callback();
+		}
+	};
+
+	function validateChannelOptions(options) {
+		if(options && 'params' in options) {
+			var params = options.params;
+			if(!Utils.isObject(params)){
+				return new ErrorInfo('options.params must be an object', 40000, 400);
+			}
+			if('modes' in params && (typeof params.modes !== 'string' || !params.modes.length)) {
+				return new ErrorInfo('options.params.modes must be a non-empty string', 40000, 400);
+			}
+			if('delta' in params && (typeof params.delta !== 'string' || !params.delta.length)) {
+				return new ErrorInfo('options.params.delta must be a non-empty string', 40000, 400);
+			}
```

Comment:
Picking two params that we happen to support now and validating them client-side doesn't really fit with how we decided to treat the params object (as an opaque bunch of key-value pairs that the client library shouldn't be looking into, just passing them on to realtime), I think we should just get rid of these

Reply:
Fixed already.